rotkehlchen/data_import/importers/bitcoin_tax.py

```diff
@@ -98,42 +99,21 @@ def _consume_trade_event(
             receive_asset_balance = quote_asset_balance
             spend_asset_balance = base_asset_balance
 
-        spend_event = HistoryEvent(
-            event_identifier=event_identifier,
-            sequence_index=0,
-            timestamp=timestamp,
-            location=location,
-            asset=spend_asset_balance.asset,
-            amount=spend_asset_balance.balance.amount,
-            notes=memo,
-            event_type=HistoryEventType.TRADE,
-            event_subtype=HistoryEventSubType.SPEND,
-        )
-        receive_event = HistoryEvent(
-            event_identifier=event_identifier,
-            sequence_index=1,
-            timestamp=timestamp,
-            location=location,
-            asset=receive_asset_balance.asset,
-            amount=receive_asset_balance.balance.amount,
-            notes=memo,
-            event_type=HistoryEventType.TRADE,
-            event_subtype=HistoryEventSubType.RECEIVE,
-        )
-        self.add_history_events(write_cursor, [spend_event, receive_event])
-        if fee_asset_balance is not None:
-            fee_event = HistoryEvent(
+        self.add_history_events(
+            write_cursor=write_cursor,
+            history_events=create_swap_events(
                 event_identifier=event_identifier,
-                sequence_index=2,
                 timestamp=timestamp,
                 location=location,
-                asset=fee_asset_balance.asset,
-                amount=fee_asset_balance.balance.amount,
-                notes=memo,
-                event_type=HistoryEventType.TRADE,
-                event_subtype=HistoryEventSubType.FEE,
-            )
-            self.add_history_events(write_cursor, [fee_event])
+                spend_asset=spend_asset_balance.asset,
+                receive_asset=receive_asset_balance.asset,
+                spend_amount=spend_asset_balance.amount,
+                receive_amount=receive_asset_balance.amount,
+                spend_notes=memo,
+                fee_asset=fee_asset_balance.asset if fee_asset_balance is not None else None,  # type: ignore[arg-type]
+                fee_amount=fee_asset_balance.amount if fee_asset_balance is not None else None,  # type: ignore[arg-type]
```

Comment:
here you are checking twice for the `fee_asset_balance` to check if it is none or not. You can do something like

```
fee_asset, fee_amount = (fee_asset_balance.asset, fee_asset_balance.amount) if fee_asset_balance is not None is not None else (None, None)
```